subsys/caf/modules/sensor_event_aggregator.c

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <drivers/sensor.h>
+#include <event_manager.h>
+
+#include <caf/events/sensor_event.h>
+#include <caf/events/sensor_aggregator_event.h>
+#include <caf/sensor_manager.h>
+
+#define MODULE sensor_event_aggregator
+#include <caf/events/module_state_event.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_CAF_SENSOR_MANAGER_LOG_LEVEL);
+
+#define DT_DRV_COMPAT caf_aggregator
+
+#define __DEFINE_DATA(i, agg_id, size)							\
+	static uint8_t agg_ ## agg_id ## _buff_ ## i ## _data[size];
+
+#define __INITIALIZE_BUF(i, agg_id, size)						\
+	{ (uint8_t *) &agg_ ## agg_id ## _buff_ ## i ## _data, size},
+
+#define __DEFINE_BUF_DATA(i)								\
+	UTIL_LISTIFY(DT_INST_PROP(i, number_of_buf), __DEFINE_DATA, i,			\
+				DT_INST_PROP(i, buf_data_length))			\
+	static struct aggregator_buffer agg_ ## i ## _bufs[] = {			\
+	UTIL_LISTIFY(DT_INST_PROP(i, number_of_buf), __INITIALIZE_BUF, i,		\
+				DT_INST_PROP(i, buf_data_length))			\
+	};
+
+#define __DEFINE_AGGREGATOR(i)								\
+	[i].descr = DT_INST_PROP(i, descr),						\
+	[i].sensor_data_size = DT_INST_PROP(i, sensor_data_size),			\
+	[i].number_of_buf = DT_INST_PROP(i, number_of_buf),				\
+	[i].agg_buffers = (struct aggregator_buffer *) &agg_ ## i ## _bufs,
+
+#define __DEFINE_AGG							\
+	static struct aggregator aggregator_arr[] = {			\
+	DT_INST_FOREACH_STATUS_OKAY(__DEFINE_AGGREGATOR)		\
+	}
+
+#define DEFINE_AGGREGATORS					\
+	DT_INST_FOREACH_STATUS_OKAY(__DEFINE_BUF_DATA)		\
+	__DEFINE_AGG
+
+#define AGG_BUF_READY 0
+#define	AGG_BUF_OCCUPIED 1
+
+struct aggregator_buffer {
+	uint8_t *data;			/* Dynamic data. */
+	uint8_t data_len;		/* Size of single sensor data in bytes. */
+	bool buffer_status;		/* Buffer status. */
```

Comment:
s/buffer_status/busy
busy means either active (is being filled in but not yet sent), or sent out
or to match what you have now busy means it is sent out and non-busy can be active or empty and waiting